pkg/repository/repository.go

```diff
@@ -45,10 +45,19 @@ type repositoryURL struct {
 	url.URL
 }
 
-// String is a custom implementation for the url where User is removed
+// String is a custom implementation for the url where User is removed and the schema is forced to https
 func (r repositoryURL) String() string {
 	repo := r.URL
 	repo.User = nil
+
+	switch repo.Scheme {
```

Comment:
I don't think we should change this function or at least we should change the name of the function. We are assuming that in all the places where we want the string representation of a repository URL we want to use `https` protocol. In some cases we might just want the URL as specified by the user.

Among other things, this is also being used by the function `GetAnonymizedRepo`, that is being used in different places. Have we double checked that in all the places where that is being used doesn't matter to change the URL scheme?

Reply:
I have double checked. We use `GetAnonymizedRepo` in:
- build command for sending analytics, so with this change we'll be sending always the same scheme for the repo and have better metrics
- pipeline command in order to show a list of the repositories, which I think we should be showing all with the same schema also.
- In our info/debug logs to hide user information
- The new insights package

The only one that I have doubts is with the logs. It's only used in the inferer, but given that the comparison between two repositories URL takes that into account I think we are good to go